src/NativeScript/ObjC/TNSArrayAdapter.mm

```diff
@@ -48,4 +48,38 @@ - (id)objectAtIndex:(NSUInteger)index {
     return toObject(self->_execState, self->_object.get()->get(self->_execState, index));
 }
 
+- (NSUInteger)countByEnumeratingWithState:(NSFastEnumerationState *)state objects:(id [])buffer count:(NSUInteger)len {
+    if (state->state == 0) { // uninitialized
+        if (jsDynamicCast<JSArray*>(self->_object.get()) != nullptr) {
+            state->state = 1;
+            state->mutationsPtr = reinterpret_cast<unsigned long*>(self);
+            state->extra[0] = 0; // current index
+        } else {
+            state->state = 2;
+        }
+    }
+
+    if (state->state == 2) { // we don't want to bother with bulk enumeration for regular objects right now, so fall back to super
+        return [super countByEnumeratingWithState:state objects:buffer count:len];
```

Comment:
I'll suggest to move this check at the top, because modifying the state will break the super logic.